app/src/main/java/edu/mit/mitmobile2/tour/fragment/TourStopsFragment.java

```diff
@@ -44,15 +54,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
         tour = getArguments().getParcelable(Constants.TOURS);
 
-        mainLoopStops = tour.getStops();
-        Iterator<MITTourStop> iterator = mainLoopStops.iterator();
-        while (iterator.hasNext()) {
-            if (!iterator.next().getType().equals(Constants.MAIN_LOOP)) {
-                iterator.remove();
-            }
-        }
+        mainLoopStops =  TourUtils.getMainLoopStops(tour.getStops());
 
-        tourStopViewPagerAdapter = new TourStopViewPagerAdapter(getFragmentManager(), mainLoopStops);
+        tourStopViewPagerAdapter = new TourStopViewPagerAdapter(getFragmentManager(), tour);
         tourStopViewpager.setAdapter(tourStopViewPagerAdapter);
         tourStopViewpager.setCurrentItem(getArguments().getInt(Constants.CURRENT_MAIN_LOOP_STOP));
```

Comment:
You're going to want to set the current item in the middle of the infinite view pager to make it swipeable either way from the start.